geode-core/src/main/java/org/apache/geode/internal/process/MBeanProcessController.java

```diff
@@ -260,6 +265,7 @@ private JMXServiceURL getJMXServiceURL() throws AttachNotSupportedException, IOE
         // need to load the management-agent and get the address
 
         final String javaHome = vm.getSystemProperties().getProperty("java.home");
+        assertState(StringUtils.isNotBlank(javaHome), CliStrings.JAVA_HOME_NOT_FOUND_ERROR_MESSAGE);
```

Comment:
I worry about changing the exception the exception thrown here (before I think it would have thrown an `IOException`).